src/Components/Web.JS/src/Boot.Server.ts

```diff
@@ -97,7 +97,7 @@ async function initializeConnection(options: CircuitStartOptions, logger: Logger
 
   connection.on('JS.AttachComponent', (componentId, selector) => attachRootComponentToLogicalElement(0, circuit.resolveElement(selector), componentId));
   connection.on('JS.BeginInvokeJS', DotNet.jsCallDispatcher.beginInvokeJSFromDotNet);
-  connection.on('JS.EndInvokeDotNet', (args: string) => DotNet.jsCallDispatcher.endInvokeDotNetFromJS(...(JSON.parse(args) as [string, boolean, unknown])));
+  connection.on('JS.EndInvokeDotNet', (args: string) => DotNet.jsCallDispatcher.endInvokeDotNetFromJS(...(DotNet.parseJsonWithRevivers(args) as [string, boolean, unknown])));
```

Comment:
This change was necessary to fix an obscure bug that I don't think had been caught until now.

On Blazor WebAssembly, if you return a `DotNetObjectReference` or `JSObjectReference` from a `DotNet.invokeMethodAsync` call, it gets properly revived in JS. However, on Blazor Server, and for async calls only, no such revival occurred.

If this change needs to be reverted for some reason, I'd be fine doing that, but this seems like a scenario we should support, especially since it already works in wasm.